testng-core/src/main/java/org/testng/internal/invokers/TestMethodWorker.java

```diff
@@ -221,6 +218,10 @@ protected void invokeAfterClassMethods(ITestClass testClass, IMethodInstance mi)
               .build();
       m_configInvoker.invokeConfigurations(attributes);
     }
+
+    for (IClassListener listener : m_listeners) {
```

Comment:
It looks like there have been no tests that checks for this (which perhaps explains how the build was succesful). Can we please add a new test that explicitly tests out this as well ?

Reply:
Added. Let me know what you think and/or what I messed up!